src/workspace.coffee

```diff
@@ -160,6 +161,13 @@ class Workspace extends Model
       @activeItemSubscriptions.add(titleSubscription) if titleSubscription?
       @activeItemSubscriptions.add(modifiedSubscription) if modifiedSubscription?
 
+  subscribeToAddedItems: ->
+    @onDidAddPaneItem ({item, pane, index}) =>
+      if item instanceof TextEditor
+        grammarSubscription = item.observeGrammar(@handleGrammarUsed.bind(this))
+        item.onDidDestroy -> grammarSubscription.dispose()
+        @emitter.emit 'did-add-text-editor', {textEditor: item, pane, index}
+
```

Comment:
It seems like we need to look at every editor being added to the workspace in order to make sure we always trigger the `grammar-used` hooks.